odoo/cli/tsconfig.py

```diff
@@ -0,0 +1,130 @@
+        return self.clean_path(f"{prefix}/{path}/{suffix}")
+
+    def remove_(self, modules, module):
+        for name, path in modules:
+            if module == name:
+                modules.remove((name, path))
+
+    def run(self, cmdargs):
+        parser = argparse.ArgumentParser(
+            prog="%s %s" % (sys.argv[0].split(os.path.sep)[-1], self.command_name),
+            description=self.__doc__
+        )
+        parser.add_argument('--addons-path', type=str, nargs=1, dest="paths")
+        args = parser.parse_args(args=cmdargs)
+
+        modules = {}
+        for path in args.paths[0].split(','):
+            for module in self.get_module_list(self.clean_path(path)):
+                modules[module] = self.prefix_suffix_path(module, path, "/static/src/*")
+
+        content = self.generate_file_content(modules)
+        # pylint: disable=bad-builtin
+        print(json.dumps(content, indent=2))
+
+    def generate_imports(self, modules):
+        return {
+            f'@{module}/*': [path]
+            for module, path in modules.items()
+        }
+
+    def generate_file_content(self, modules):
+        return {
+            'compilerOptions': {
+                "baseUrl": ".",
+                "checkJs": True,
+                "allowJs": True,
+                "noEmit": True,
+                "paths": self.generate_imports(modules)
+            }, "exclude": self.generate_excludes()
+        }
+
+    def generate_excludes(self):
+        return [
+            "/**/*.po",
+            "/**/*.py",
+            "/**/*.pyc",
+            "/**/*.xml",
+            "/**/*.png",
+            "/**/*.md",
+            "/**/*.dat",
+            "/**/*.scss",
+            "/**/*.jpg",
+            "/**/*.svg",
+            "/**/*.pot",
+            "/**/*.csv",
+            "/**/*.mo",
+            "/**/*.txt",
+            "/**/*.less",
+            "/**/*.bcmap",
+            "/**/*.properties",
+            "/**/*.html",
+            "/**/*.ttf",
+            "/**/*.rst",
+            "/**/*.css",
+            "/**/*.pack",
+            "/**/*.idx",
+            "/**/*.h",
+            "/**/*.map",
+            "/**/*.gif",
+            "/**/*.sample",
+            "/**/*.doctree",
+            "/**/*.so",
+            "/**/*.pdf",
+            "/**/*.xslt",
+            "/**/*.conf",
+            "/**/*.woff",
+            "/**/*.xsd",
+            "/**/*.eot",
+            "/**/*.jst",
+            "/**/*.flow",
+            "/**/*.sh",
+            "/**/*.yml",
+            "/**/*.pfb",
+            "/**/*.jpeg",
+            "/**/*.crt",
+            "/**/*.template",
+            "/**/*.pxd",
+            "/**/*.dylib",
+            "/**/*.pem",
+            "/**/*.rng",
+            "/**/*.xsl",
+            "/**/*.xls",
+            "/**/*.cfg",
+            "/**/*.pyi",
+            "/**/*.pth",
+            "/**/*.markdown",
+            "/**/*.key",
+            "/**/*.ico",
+        ]
```

Comment:
Is there no way to specify things the other way around, including the files we want rather than excludnkg everything else?

Reply:
@xmo-odoo  I tried it. I also tried both together, by excluding everything and including only what is needed. It is a bit tricky because each IDE parse and understand this file slightly differently. This method does rather well on vscode and pycharm so I kept it. However, after feedbacks after deployments, I might change it again. Maybe even add a flag in the generation command to ask for a specific IDE so the generation is more suited.

Reply:
That seems fair enough.